File: test/aidbox_sdk/snapshots/typescripthl7-fhir-r4-core/SpecimenDefinition.ts
import { CodeableConcept } from "./CodeableConcept";
import { Range } from "./Range";
import { Quantity } from "./Quantity";
import { Duration } from "./Duration";
import { DomainResource } from "./DomainResource";
import { Reference } from "./Reference";
import { Identifier } from "./Identifier";
import { BackboneElement } from "./BackboneElement";

export type SpecimenDefinitionTypeTestedHandling = BackboneElement & {
    instruction?: string;
    maxDuration?: Duration;
    temperatureRange?: Range;
    temperatureQualifier?: CodeableConcept;
};

export type SpecimenDefinitionTypeTestedContainerAdditive = BackboneElement & {
    additive?: Reference | CodeableConcept;
};

export type SpecimenDefinitionTypeTestedContainer = BackboneElement & {
    description?: string;
    capacity?: Quantity;
    type?: CodeableConcept;
    cap?: CodeableConcept;
    preparation?: string;
    minimumVolume?: Quantity | string;
    material?: CodeableConcept;
    additive?: SpecimenDefinitionTypeTestedContainerAdditive[];
};

export type SpecimenDefinitionTypeTested = BackboneElement & {
    type?: CodeableConcept;
    handling?: SpecimenDefinitionTypeTestedHandling[];
    container?: SpecimenDefinitionTypeTestedContainer;
    isDerived?: boolean;
    preference: string;
    requirement?: string;
    retentionTime?: Duration;
    rejectionCriterion?: CodeableConcept[];
};

export type SpecimenDefinition = DomainResource & {
    collection?: CodeableConcept[];
    identifier?: Identifier;
    timeAspect?: string;
    typeTested?: SpecimenDefinitionTypeTested[];
    typeCollected?: CodeableConcept;
    patientPreparation?: CodeableConcept[];
};

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/SpecimenDefinition.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import CodeableConcept
from base import Range
from base import Quantity
from base import Duration
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class SpecimenDefinitionTypeTestedHandling(BackboneElement):
    instruction: Optional[str] = None
    max_duration: Optional[Duration] = None
    temperature_range: Optional[Range] = None
    temperature_qualifier: Optional[CodeableConcept] = None

@dataclass(kw_only=True)
class SpecimenDefinitionTypeTestedContainerAdditive(BackboneElement):
    additive_reference: Optional[Reference] = None
    additive_codeable_concept: Optional[CodeableConcept] = None

@dataclass(kw_only=True)
class SpecimenDefinitionTypeTestedContainer(BackboneElement):
    description: Optional[str] = None
    capacity: Optional[Quantity] = None
    minimum_volume_quantity: Optional[Quantity] = None
    type: Optional[CodeableConcept] = None
    cap: Optional[CodeableConcept] = None
    preparation: Optional[str] = None
    material: Optional[CodeableConcept] = None
    additive: Optional[List[SpecimenDefinitionTypeTestedContainerAdditive]] = field(default_factory=list)
    minimum_volume_string: Optional[str] = None

@dataclass(kw_only=True)
class SpecimenDefinitionTypeTested(BackboneElement):
    type: Optional[CodeableConcept] = None
    handling: Optional[List[SpecimenDefinitionTypeTestedHandling]] = field(default_factory=list)
    container: Optional[SpecimenDefinitionTypeTestedContainer] = None
    is_derived: Optional[bool] = None
    preference: str
    requirement: Optional[str] = None
    retention_time: Optional[Duration] = None
    rejection_criterion: Optional[List[CodeableConcept]] = field(default_factory=list)

@dataclass(kw_only=True)
class SpecimenDefinition(DomainResource):
    collection: Optional[List[CodeableConcept]] = field(default_factory=list)
    identifier: Optional[Identifier] = None
    time_aspect: Optional[str] = None
    type_tested: Optional[List[SpecimenDefinitionTypeTested]] = field(default_factory=list)
    type_collected: Optional[CodeableConcept] = None
    patient_preparation: Optional[List[CodeableConcept]] = field(default_factory=list)